engine/clients/pgvector/upload.py

```diff
@@ -27,10 +32,28 @@ def upload_batch(
         vectors = np.array(vectors)
 
         # Copy is faster than insert
-        with cls.cur.copy("COPY items (id, embedding) FROM STDIN") as copy:
+        with cls.cur.copy(
+            "COPY items (id, embedding) FROM STDIN WITH (FORMAT BINARY)"
+        ) as copy:
+            copy.set_types(["integer", "vector"])
             for i, embedding in zip(ids, vectors):
                 copy.write_row((i, embedding))
 
+    @classmethod
+    def post_upload(cls, distance):
+        try:
+            hnsw_distance_type = cls.DISTANCE_MAPPING[distance]
+        except KeyError:
+            raise IncompatibilityError(f"Unsupported distance metric: {distance}")
+
+        cls.conn.execute("SET max_parallel_workers = 128")
+        cls.conn.execute("SET max_parallel_maintenance_workers = 128")
```

Comment:
This will use all available cores to build the index (let me know if it should do something different)

Reply:
If you want to utilize all cores, would not it be better to get the number of cores from OS rather than to hardcode it to 128?

Reply:
Setting a high upper limit should be simpler than trying to get the exact number of cores from the server (and won't make a difference in the number of cores used).